store/sqlstore/integrity.go

```diff
@@ -0,0 +1,504 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"fmt"
+
+	"github.com/mattermost/gorp"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type relationalCheckConfig struct {
+	parentName         string
+	parentIdAttr       string
+	childName          string
+	childIdAttr        string
+	canParentIdBeEmpty bool
+	sortRecords        bool
+}
+
+func getOrphanedRecords(dbmap *gorp.DbMap, cfg relationalCheckConfig) ([]store.OrphanedRecord, error) {
```

Comment:
Is there a reason to accept a `*gorp.DbMap` instead of just a `SqlStore`?

Reply:
Being new to the codebase I wasn't sure what the best practice was. I looked at the code and saw that `*gorp.DbMap` was the main entry point to "speak" to the DB. Since I had to run queries I thought to pass the only struct I actually needed.

Reply:
In my mind, `*gorp.DbMap` is an implementation detail to which we'd ideally limit exposure. There's a decent chance that gorp goes away at some point in the future, and having code that isn't explicitly dependent on it will make that migration easier.

Reply:
I understand your reasoning. Do you suggest I pass the `SqlSupplier` and call `GetMaster()` right before running the query (which means using gorp just in `getOrphanedRecords`) or is there a different way to run custom sql I am not aware of?

Reply:
1/5 if possible, but I wouldn't press too hard on this :)